plugins/source-terasology-modules/gatsby-node.js

```diff
@@ -92,78 +105,118 @@ exports.sourceNodes = async (
 
     await cache.set(dataKey, JSON.stringify(repositories));
     await cache.set(lastFetchedKey, now.toISO());
+
+    fs.writeFileSync(
+      `${__dirname}/data.json`,
+      JSON.stringify(repositories, null, 2)
+    );
   }
 
   reporter.success(`[${PLUGIN_NAME}] Loaded ${repositories.length} modules.`);
 
-  let created = 0;
-  repositories
-    .forEach((repo) => {
-      if (!repo.moduleTxt) {
-        // skip non-module repositories
-        return;
-      }
+  const nodes = repositories.flatMap(async (repo) => {
+    if (!repo.moduleTxt) {
+      // skip non-module repositories
+      return [];
+    }
 
-      let moduleTxt;
+    let moduleTxt;
+    try {
+      moduleTxt = JSON.parse(repo.moduleTxt?.text);
+    } catch (err) {
+      console.warn(
+        `[${PLUGIN_NAME}] Could not parse 'module.txt' of ${repo.url}.`
+      );
+      return [];
+    }
+
+    const tags = [
+      moduleTxt.isTutorial ? "Tutorial" : undefined,
+      moduleTxt.isAsset ? "Asset" : undefined,
+      moduleTxt.isLibrary ? "Library" : undefined,
+      moduleTxt.isSpecial ? "Special" : undefined,
+      moduleTxt.isRendering ? "Rendering" : undefined,
+      moduleTxt.isAugmentation ? "Augmentation" : undefined,
+      moduleTxt.isServerSideOnly ? "Server" : undefined,
+      moduleTxt.isGameplay ? "Gameplay" : undefined,
+    ].filter((x) => x);
+
+    const { id, version, displayName, description, dependencies } = moduleTxt;
+
+    const cover = repo.usesCustomOpenGraphImage
+      ? repo.openGraphImageUrl
+      : undefined;
+
+    const moduleInfo = {
+      id,
+      version,
+      displayName,
+      description,
+      dependencies,
+      tags,
+    };
+
+    moduleInfo.dependencies = moduleInfo.dependencies.map((dep) => ({
+      ...dep,
+      optional: /true/i.test(dep.optional),
```

Comment:
can we test for the full `"optional": true` here to harden against additional fields being added to the dependency entries?

Reply:
I'm not sure I get what you want here :thinking: This weird regex test is to turn something that is either a boolean or a string that looks like a boolean into an actual boolean (GraphQL complains as it cannot derive a proper type otherwise).

```
true 		-> true
false 		-> false
"true" 	-> true
"True" 	-> true
"false"	-> false
"foo"		-> false
```

Reply:
oh then I didn't understand correctly what it did. I thought you are checking whether or not the dependency is optional by checking whether the `"optional": true` line exists. Nevermind then.